sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -577,9 +577,11 @@ object SQLConf {
     .doc("The maximum lines of a single Java function generated by whole-stage codegen. " +
       "When the generated function exceeds this threshold, " +
       "the whole-stage codegen is deactivated for this subtree of the current query plan. " +
-      "The default value 4000 is the max length of byte code JIT supported " +
-      "for a single function(8000) divided by 2.")
+      "The default value 2667 is the max length of byte code JIT supported " +
+      "for a single function(8000) divided by 2. Use -1 to disable this.")
     .intConf
+    .checkValue(maxLines => maxLines >= -1, "The maximum must not be a negative integer, -1 to " +
+      "always activate whole-stage codegen.")
```

Comment:
`The maximum must not be a negative integer, except for -1 using to always activate whole-stage codegen.`